R/expect_identical_linter.R

```diff
@@ -69,6 +69,11 @@ expect_identical_linter <- function() {
         and expr[NUM_CONST[contains(text(), '.')]]
```

Comment:
Hmm I saw this too -- so we haven't fully restored parity yet in this PR.

```r
expect_equal(x, 1.02)     # no lint
expect_equal(x, c(1.02))  # no lint
expect_equal(x, -1.02)    # no lint
expect_equal(x, c(-1.02)) # lint!
```

Let's at least get parity in this PR. I would put all of the conditions on `following-sibling::expr[]` together btw, like:

```xpath
following-sibling::expr[
  (
    expr[1]/SYMBOL_FUNCTION_CALL[text() = 'c']
    and expr/NUM_CONST[contains(text(), '.')]
  ) or (
    NUM_CONST[contains(text(), '.')]
  ) or (
    OP-MINUS ...
  )
]
```

Since `NUM_CONST[contains(text(), '.')]` is getting repeated so much let's extract it to a variable too to save space.

Reply:
BTW, could you also test if the following lints?

```r
expect_equal(x, c(1.02, y))
```

I think it's something to possible address in follow-up. I'm wondering if maybe it was intentional not to lint `c(1.02, y)` because there's _some_ part of the expectation that uses an "exact" number so we "should" skip linting on the same principle as just `1.02`? In which case we shouldn't lint `1.01-y` either 🤔

@AshesITR any thoughts? This cut-out does feel a bit arbitrary, probably by necessity at some level. Wondering how to close the loop here for this PR, and what can be deferred to a more holistic follow-up.

Reply:
``` r
lintr::lint(
  text = "expect_equal(x, c(1.02, y))", 
  linters = lintr::expect_identical_linter()
)
```

<sup>Created on 2023-12-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

It doesn't lint.